Fix clippy warnings

MichaelMcDonnell opened the pull request:
The current clippy warnings are causing the GitHub Actions build to fail. The rule has been renamed.

MichaelMcDonnell commented:
Looks like there are still two warnings left.

TimonPost commented:
Thanks! 

Its about the Attribute enum and Error enum

https://github.com/crossterm-rs/crossterm/blob/ddc25d4225dd8e02d485000bd173a8978a2ed022/src/error.rs#L23

https://github.com/crossterm-rs/crossterm/blob/ddc25d4225dd8e02d485000bd173a8978a2ed022/src/style/types/attribute.rs#L143

If you see the opportunity to remove them feel free to do so.

MichaelMcDonnell commented:
I've updated the pull request. I fixed the non_exhaustive attribute warnings but then I got another warning about matches. I fixed that one too. There are no warning left now.

TimonPost commented:
Awesome and thanks!
